File: ts/medium/longestCommonPrefix.ts
export const longestComPrefix = (strArr: string[]): string => {
  let result = '';
  if (
    Array.isArray(strArr) &&
    strArr.length > 1 &&
    strArr.every((item) => typeof item === 'string')
  ) {
    const preparedArr: string[] = strArr.map((word) =>
      word.trim().toLowerCase(),
    );
    const firstWord: string = preparedArr.shift()!; //  not in the array anymore
    let counter = 0;

    while (counter <= firstWord.length) {
      let candidateChar = firstWord[counter];
      for (const word of preparedArr) {
        if (word[counter] !== candidateChar) {
          return result;
        }
      }
      result += candidateChar;
      counter++;
    }
  }
  return result;
};
console.log(longestComPrefix(['flower', 'flow', 'Flight']));

File: python/medium/longest_common_prefix.py
def longest_common_prefix(arr):
  result = ''
  if isinstance(arr, list) and len(arr) > 1:
    prepared_arr = []

    for item in arr:
      if isinstance(item, str):
        prepared = item.lower()
        prepared_arr.append(prepared)

    first_word = prepared_arr[0]
    prepared_arr.pop(0) 
    counter = 0

    while counter < len(first_word):
      candidate = first_word[counter]

      for word in prepared_arr:
        if word[counter] != candidate:
          return result
      
      result = result + candidate
      counter = counter + 1


  return result
print(longest_common_prefix(['flower', 'flow', 'Flight']))